openmetadata-ui/src/main/resources/ui/src/components/common/FormBuilder/FormBuilder.tsx

```diff
@@ -98,38 +96,24 @@ const FormBuilder: FunctionComponent<Props> = ({
   };
 
   const handleSubmit = () => {
-    if (formRef.current) {
+    if (formRef.current && formRef.current?.validateForm()) {
       formRef.current.submit();
     }
   };
 
-  const handleChange = (updatedData: ConfigData) => {
-    setLocalFormData(updatedData);
-  };
-
-  const transformErrors = (errors: AjvError[]) =>
-    errors.map((error) => {
-      const fieldName = error.params.missingProperty;
-      const customMessage = `${startCase(fieldName)} is required`;
-      error.message = customMessage;
-
-      return error;
-    });
-
   const handleRequiredFieldsValidation = () => {
-    const validationObject = validateFormData(localFormData, schema);
-    const isFormValid = isEmpty(validationObject.errors);
-    if (!isFormValid) {
-      formRef.current?.submit();
-    }
+    return formRef.current?.validateForm() ?? false;
+  };
 
-    return isFormValid;
+  const handleFormChange = (e: IChangeEvent) => {
+    setLocalFormData(e.formData);
```

Comment:
```suggestion
    e.formData && setLocalFormData(e.formData);
```

Reply:
Don't think we need that check, Any specific reason to add that?

Reply:
formData can be undefined, so we need to check that as well.

Reply:
setLocalFormData accepts undefined as well.